eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.test.js

```diff
@@ -60,51 +60,95 @@ describe("Row", () => {
   });
 
-  it("should not re-render if the props we care about dont change", () => {
-    const wrapper = shallow(
-      <Row
-        questionnaire={questionnaire}
-        onDeleteQuestionnaire={handleDeleteQuestionnaire}
-        onDuplicateQuestionnaire={handleDuplicateQuestionnaire}
-      />
-    );
-    expect(
-      wrapper.instance().shouldComponentUpdate(
-        {
-          questionnaire,
-          onDeleteQuestionnaire: handleDeleteQuestionnaire,
-          onDuplicateQuestionnaire: handleDuplicateQuestionnaire,
-          foo: "bar",
-        },
-        { showDeleteQuestionnaireDialog: false }
-      )
-    ).toBeFalsy();
+  it("should call push when link is clicked", () => {
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+    const tableCell = wrapper.find(TD).first();
+    tableCell.simulate("click");
+    expect(props.history.push).toHaveBeenCalled();
+  });
+
+  it("should only grab focus once", () => {
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+
+    const focus = jest.fn();
+    const getElementsByTagName = jest.fn(() => [{ focus }]);
+    const instance = wrapper.instance();
+    instance.rowRef = { current: { getElementsByTagName } };
+
+    wrapper.setProps({
+      autoFocus: true,
+    });
+
+    wrapper.setProps({
+      autoFocus: true,
+    });
+
+    expect(focus).toHaveBeenCalledTimes(1);
+  });
+
+  it("should allow duplication of a Questionnaire", () => {
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+
+    wrapper.find(DuplicateButton).simulate("click");
+
+    expect(props.onDuplicateQuestionnaire).toHaveBeenCalledWith(questionnaire);
   });
 
   it("should call delete confirm dialog when the delete button is clicked", () => {
-    const wrapper = shallow(
-      <Row
-        questionnaire={questionnaire}
-        onDeleteQuestionnaire={handleDeleteQuestionnaire}
-        onDuplicateQuestionnaire={handleDuplicateQuestionnaire}
-      />
-    );
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
     wrapper.find(IconButtonDelete).simulate("click");
 
-    expect(handleDeleteQuestionnaire).toHaveBeenCalledWith(questionnaire);
+    expect(props.onDeleteQuestionnaire).toHaveBeenCalledWith(questionnaire);
+  });
+
+  it("should render short title correctly", () => {
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+    const shortTitle = wrapper.find(ShortTitle);
+    expect(shortTitle).toMatchSnapshot();
+  });
+
+  it("should call focusLink in componentDidMount when autoFocus is true", () => {
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+    wrapper.instance().focusLink = jest.fn();
+
+    wrapper.setProps({
+      autoFocus: true,
+    });
+
+    expect(wrapper.instance().focusLink).toHaveBeenCalled();
+  });
+
+  it("should apply alignment styles to table cell", () => {
```

Comment:
Not sure what this test is for? It doesn't test any feature of this component but rather the DOM.

Reply:
Removed this test, and changed the TD to have no alignment set as the default is left.